### PR TITLE
Migration downloads individual migration binaries

### PR DESCRIPTION
**Pull request by gammazero**
When performing a migration after installing a new ipfs version, ipfs-update finds the necessary mirgations, downloads the latest version of them from the distribution site, unpacks the executables, and runs the migrations in order.  It uses the code located in `go-ipfs/repo/fsrepo/migrations` to do this.

Also uses `go-ipfs/repo/fsrepo/migrations` for:
- fetching binaries for things other than migrations
- fetching version information about binaries other than migrations
- looking at current repo version
- checking for local ipfs api availability

This PR depends on the following:
- ipfs/distributions#327
- ipfs/go-ipfs#7857

### CODE REVIEW

**Review by aschmahmann — CHANGES_REQUESTED** [review]

File: lib/install.go [hunk 1]
```diff
@@ -92,91 +91,91 @@ func (i *Install) Run() error {
...
+func (i *Install) maybeStash() error {
+	if i.currentVers != "none" {
 		stump.Log("stashing old binary")
-		oldpath, err := StashOldBinary(i.CurrentVers, false)
+		oldpath, err := StashOldBinary(i.currentVers, false)
 		if err != nil {
 			if strings.Contains(err.Error(), "could not find old") {
 				stump.Log("stash failed, no binary found.")
-				stump.Log(util.BoldText("this could be because you have a daemon running, but no ipfs binary in your path."))
+				stump.Log("** this could be because you have a daemon running, but no ipfs binary in your path. **")
```
Comment (aschmahmann) [inline lib/install.go:157]: Why this change? Didn't like the one off use of control characters here?

Comment (gammazero) [reply lib/install.go:157]: Escape characters do not parse well in logs or display on some types of terminals.

In a conversation with @Stebalien he mentioned that we probably should not be using stump or other non-standard loggers anywhere.

Comment (aschmahmann) [reply lib/install.go:157]: fair enough, control characters seem like they should probably be used only if there's a bunch of value (e.g. progress bars, etc.)

File: main.go [hunk 2]
```diff
@@ -97,30 +100,30 @@ var cmdVersions = cli.Command{
...
 		return nil
 	},
 }
 
-var cmdInstall = cli.Command{
+var cmdInstall = &cli.Command{
 	Name:      "install",
 	Usage:     "Install a version of ipfs.",
-	ArgsUsage: "A version or \"latest\" for latest version",
+	ArgsUsage: "A version or \"latest\" or \"latest-stable\" for latest version",
```
[the remark this answers is not in the document]
Comment (gammazero) [reply main.go]: done

File: main.go [hunk 3]
```diff
@@ -97,30 +100,30 @@ var cmdVersions = cli.Command{
...
 		return nil
 	},
 }
 
-var cmdInstall = cli.Command{
+var cmdInstall = &cli.Command{
 	Name:      "install",
 	Usage:     "Install a version of ipfs.",
-	ArgsUsage: "A version or \"latest\" for latest version",
+	ArgsUsage: "A version or \"latest\" for latest stable version or \"beta\" for latest or RC",
```
Comment (aschmahmann) [inline main.go:120]:
```suggestion
	ArgsUsage: "A version or \"latest\" for the latest stable version or \"beta\" for the latest stable or RC version",
```

File: main.go [hunk 4]
```diff
@@ -235,62 +235,64 @@ var cmdRevert = cli.Command{
 	},
 }
 
-var cmdFetch = cli.Command{
+var cmdFetch = &cli.Command{
 	Name:      "fetch",
-	Usage:     "Fetch a given version of ipfs. Default: latest.",
+	Usage:     "Fetch a given version of ipfs, or \"latest\" for latest stable or \"beta\" for latest or RC. Default: latest.",
```
Comment (aschmahmann) [inline main.go:240]:
```suggestion
	Usage:     "Fetch a given version of ipfs, or \"latest\" for the latest stable version or \"beta\" for the latest stable or RC version. Default: latest.",
```
